node/lib/cmd/pull.js

```diff
@@ -78,11 +78,47 @@ used if not specified`,
 };
 
 /**
+ * Return true if the user has requested a rebase (explicitly or
+ * via config).
+ *
+ * @param {Object} args
+ * @param {Boolean} args.rebase
+ * @param {Nodegit.Repository} repo
+ * @param {NodeGit.Branch} branch
+ * @async
+ * @return bool
+ */
+exports.userWantsRebase = co.wrap(function*(args, repo, branch) {
+    if (args.rebase !== undefined) {
+        return args.rebase;
+    }
+
+    const config = yield repo.config();
+    try {
+        const branchConfigVar = `branch.${branch.shorthand()}.rebase`;
+        const configured = yield config.getStringBuf(branchConfigVar);
+        return (configured === "true" || configured === "yes" ||
+                configured === "on");
+    } catch (e) {
+        // no branch config, try global config
+    }
+
+    try {
+        const configured = yield config.getStringBuf("pull.rebase");
+        return (configured === "true" || configured === "yes" ||
```

Comment:
Could you factor out the test against the three strings? It's unfortunate that libgit2 doesn't do this.